ross/defects/misalignment.py

```diff
@@ -0,0 +1,661 @@
+                0,
+                0,
+                0,
+                0,
+                +k0 * self.Kcoup_auxF * delta1 * np.sin(betam - fir),
+                k0 * self.Kcoup_auxF * delta1 * np.cos(betam - fir),
+                0,
+                0,
+                0,
+                0,
+            ]
+        )
+
+        K_mis_matrix = np.zeros((12, 12))
+        K_mis_matrix[5, :] = k_misalignbeta1
+        K_mis_matrix[11, :] = -k_misalignbeta1
+
+        Force_kkmis = K_mis_matrix.dot(positions[self.DoF])
+
+        F_misalign = np.array(
+            [
+                -(-k0 * delta1 * np.cos(betam - fir) + k0 * delta1),
+                -k0 * delta1 * np.sin(betam - fir),
+                0,
+                0,
+                0,
+                self.TD - self.TL,
+                -(k0 * delta1 * np.cos(betam - fir) - k0 * delta1),
+                k0 * delta1 * np.sin(betam - fir),
+                0,
+                0,
+                0,
+                -(self.TD - self.TL),
+            ]
+        )
+
+        Fmis = Force_kkmis + F_misalign
+        FFmis = np.zeros(self.ndof)
+        FFmis[self.DoF] = Fmis
+
+        return Fmis, FFmis
+
+    def force(self):
+        return self._parallel()
+
+    def plot_time_response(
+        self,
+        probe,
+        probe_units="rad",
+        displacement_units="m",
+        time_units="s",
+        fig=None,
+        **kwargs,
+    ):
+        """
+        """
+        if fig is None:
+            fig = go.Figure()
+
+        for i, p in enumerate(probe):
+            dofx = p[0] * self.rotor.number_dof
+            dofy = p[0] * self.rotor.number_dof + 1
+            angle = Q_(p[1], probe_units).to("rad").m
+
+            # fmt: off
+            operator = np.array(
+                [[np.cos(angle), - np.sin(angle)],
+                 [np.cos(angle), + np.sin(angle)]]
+            )
+
+            _probe_resp = operator @ np.vstack((self.response[dofx,:], self.response[dofy,:]))
+            probe_resp = (
+                _probe_resp[0] * np.cos(angle) ** 2  +
+                _probe_resp[1] * np.sin(angle) ** 2
+            )
+            # fmt: on
+
+            probe_resp = Q_(probe_resp, "m").to(displacement_units).m
+
+            fig.add_trace(
+                go.Scatter(
+                    x=Q_(self.time_vector, "s").to(time_units).m,
+                    y=Q_(probe_resp, "m").to(displacement_units).m,
+                    mode="lines",
+                    name=f"Probe {i + 1}",
+                    legendgroup=f"Probe {i + 1}",
+                    showlegend=True,
+                    hovertemplate=f"Time ({time_units}): %{{x:.2f}}<br>Amplitude ({displacement_units}): %{{y:.2e}}",
+                )
+            )
+
+        fig.update_xaxes(title_text=f"Time ({time_units})")
+        fig.update_yaxes(title_text=f"Amplitude ({displacement_units})")
+        fig.update_layout(**kwargs)
+
+        return fig
```

Comment:
I think you can call `TimeResponseResults` here, instead of repeating `.plot()` method. Also, changing the method's name would fit it better.
For example:

```python
def run_time_response():
    results = TimeResponseResults(
        t=self.time_vector,
        yout=self.response.T,    # I'm transposing the matrix to match the method's syntax,
        xout=[],                 # It doesn't matter in this case.
        nodes_list=self.rotor.nodes,
        nodes_pos=self.rotor.nodes_pos,
        number_dof=self.rotor.number_dof,
    )
    return results
```

Then, if plotting the results is required, you can make the same procedure `rotor_assembly.py` does:
```python
example = MisalignmentRigid(tI, tF, Kcoup_auxI, Kcoup_auxF, kCOUP, eCOUP, TD, TL, n1, speed)
results = example.run_time_response()
fig = results.plot_1d()  # plot_1d() is a method from TimeResponseResults class


```

Reply:
I think these changes would be interesting because you could also take advantage from `TimeResponseResults` to plot orbits.

```python
example = MisalignmentRigid(tI, tF, Kcoup_auxI, Kcoup_auxF, kCOUP, eCOUP, TD, TL, n1, speed)
results = example.run_time_response()
fig1d = results.plot_1d()  # plots results for given probes
fig2d = results.plot_2d()  # plots the orbit of a given node
fig3d = results.plot_3d()  # plots the rotor orbit (all nodes).

```